src/net_processing.cpp

```diff
@@ -1723,6 +1741,11 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                 // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
                 assert(recentRejects);
                 recentRejects->insert(tx.GetHash());
+                if (RecursiveDynamicUsage(*ptx) < 100000) {
+                    AddToCompactExtraTransactions(ptx);
+                }
+            } else if (tx.HasWitness() && RecursiveDynamicUsage(*ptx) < 100000) {
```

Comment:
Yea, this is no worse than before, and in many cases is a big win...its also written to be an easy review and merge, with the opportunity of cleaning it up and making it more clever in the future.